src/raspi/cpp/Makefile

```
@@ -0,0 +1,23 @@
+include	../.env # if this is saying that there is a seperator check the env file that any documentation is commented out
+
+ifeq ($(ENV), "dev")
+LIBRARIES= 
+DEFINITION=COMPUTER
+else
+LIBRARIES=-lpigpio -lrt
+DEFINITION=ROBOT
+endif
+
+
+output/main: main.o motor.o
+	g++ -o output/main -g output/main.o output/motor.o 
+
+main.o:
+	g++ -c -g -Wall -D $(DEFINITION) -o output/main.o main.cpp $(LIBRARIES)
+
+motor.o: io_controller/motor.cpp io_controller/motor.h
+	g++ -c -g -Wall -D COMPUTER -o output/motor.o io_controller/motor.cpp
+
```

```suggestion
ifeq ($(ENV), "dev")
DEFINITION=COMPUTER
else
DEFINITION=ROBOT
endif


output/main: main.o motor.o
	g++ -o output/main -g output/main.o output/motor.o 

main.o:
	ifeq (DEFINITION, ROBOT)
	LIBRARIES=-lpigpio -lrt
	else
	LIBRARIES=
	endif
	g++ -c -g -Wall -D $(DEFINITION) -o output/main.o main.cpp $(LIBRARIES)

motor.o: io_controller/motor.cpp io_controller/motor.h
	g++ -c -g -Wall -D $(DEFINITION) -o output/motor.o io_controller/motor.cpp

```
If this doesn't work then disregard, but I think its better to do it this way such that if other modules use different set of libraries then we can incorporate that.

---

Also does main.cpp require those libraries or does motor require those libraries or does that matter at all.